packages/builders/src/util/validation.ts

```diff
@@ -24,3 +27,20 @@ export function disableValidators() {
 export function isValidationEnabled() {
 	return validate;
 }
+
+/**
+ * Parses a value with a given validator
+ *
+ * @param validator - Tthe zod validator to use
+ * @param value - The value to parse
+ * @returns The result from parsing
+ * @internal
+ */
+export function parse<Validator extends ZodTypeAny>(validator: Validator, value: unknown): output<Validator> {
+	const result = validator.safeParse(value);
+	if (isValidationEnabled() && !result.success) {
```

Comment:
Shouldn't validationEnabled be checked before everything else? 👀

Reply:
Afaik shapeshift doesn’t do it either, because we also parse for example nativeEnums and replace them with their enum values if the string key is passed. That’s why we need the parse result even if validation is disabled.